### PR TITLE
Itinerary Body: migrate to OTP2 trip props

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
The OTP1 trip props are no longer in use, and are fundamentally incompatible with the OTP2 props. This PR resolves this incongruity.

### CODE REVIEW

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
I smell a breaking change so it will require a breaking change commit. To avoid a breaking change, fall back on `fromIndex` and `toIndex` if `fromStopId` and `toStopId` are not provided.

File: packages/itinerary-body/src/TransitLegBody/view-trip-button.tsx [hunk 1]
```diff
@@ -11,8 +11,8 @@ type Props = TripSection & {
 
 class ViewTripButton extends Component<Props> {
   onClick = (): void => {
-    const { fromIndex, setViewedTrip, toIndex, tripId } = this.props;
-    setViewedTrip({ fromIndex, toIndex, tripId });
+    const { fromStopId, setViewedTrip, toStopId, tripId } = this.props;
+    setViewedTrip({ fromStopId, toStopId, tripId });
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/view-trip-button.tsx:15]: Yes

**Comment by miles-grant-ibigroup** [comment]
I think we have to make a breaking change. OTP1 is no longer supported, so it makes no sense to support it here. I will modify the commit message

**Review by binh-dam-ibigroup — COMMENTED** [review]
Could you update the mocks for the "Individual Leg Fare Components" to have `fromStopId` and `toStopId`?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
See https://github.com/opentripplanner/otp-ui/pull/640#pullrequestreview-1595578597

**Comment by miles-grant-ibigroup** [comment]
Updating the mocks would be a 2-release process...

File: packages/itinerary-body/src/TransitLegBody/view-trip-button.tsx [hunk 2]
```diff
@@ -11,8 +11,19 @@ type Props = TripSection & {
 
 class ViewTripButton extends Component<Props> {
   onClick = (): void => {
-    const { fromIndex, setViewedTrip, toIndex, tripId } = this.props;
-    setViewedTrip({ fromIndex, toIndex, tripId });
+    const {
+      fromIndex,
+      toIndex,
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/view-trip-button.tsx:16]: This wasn't addressed.

[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/view-trip-button.tsx:16]: See 2836d0e84d502018f665127a1a20f78ed79af8ba

**Comment by binh-dam-ibigroup** [comment]
> Updating the mocks would be a 2-release process...

Why? The mocks are in the itinerary-body package.

**Comment by miles-grant-ibigroup** [comment]
> > Updating the mocks would be a 2-release process...
> 
> Why? The mocks are in the itinerary-body package.

Wait sorry, I got myself confused. Mocks are updated

**Comment by miles-grant-ibigroup** [comment]
I'm not sure what's going on... The new mocks aren't showing up in the GitHub diffs either, but they are showing up in my local git commit history

File: packages/itinerary-body/src/TransitLegBody/index.tsx [hunk 3]
```diff
@@ -321,16 +321,12 @@ class TransitLegBody extends Component<Props, State> {
 
                   {showViewTripButton && (
                     <ViewTripButton
-                      tripId={leg.tripId}
-                      fromIndex={
-                        leg.from.stopIndex ||
-                        leg?.trip?.departureStoptime?.stopPosition
-                      }
+                      fromIndex={leg.from?.stopIndex}
+                      fromStopId={leg?.from?.stopId}
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/TransitLegBody/index.tsx:325]: I think I know what is going on. The "from" and "to" stop ids come in OTP1 and OTP2 flavors. The code change in this file only cover the OTP1 response. The change you made to the mocks cover OTP2 response, which is not covered by this code, so please handle that. You need to add `leg.trip?.fromStopId` then fall back on `leg.from?.stopId`.

Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/index.tsx:325]: The problem is the actual re-writing of the response is handled by OTP-RR. I can add a `stopId` to the mock, but this wouldn't represent actual OTP2 responses. It relies on otp-rr rewriting the responses

Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/index.tsx:325]: In that case revert the mocks I made you change.

Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/index.tsx:325]: Done!

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM, sorry for the confusion!

File: packages/itinerary-body/src/TransitLegBody/index.tsx [hunk 4]
```diff
@@ -321,16 +321,12 @@ class TransitLegBody extends Component<Props, State> {
...
                       setViewedTrip={setViewedTrip}
-                      toIndex={
-                        leg.to.stopIndex ||
-                        leg?.trip?.arrivalStoptime?.stopPosition
-                      }
+                      toIndex={leg.to?.stopIndex}
+                      toStopId={leg?.to?.stopId}
```
Comment (daniel-heppner-ibigroup) [inline packages/itinerary-body/src/TransitLegBody/index.tsx:328]: unneeded question mark

File: packages/itinerary-body/src/TransitLegBody/index.tsx (hunk 3 above)
Comment (daniel-heppner-ibigroup) [inline packages/itinerary-body/src/TransitLegBody/index.tsx:325]: unneded question mark

**Review by daniel-heppner-ibigroup — APPROVED** [review]
I think the types could use some work, but this looks good... Thanks!

File: packages/itinerary-body/src/TransitLegBody/index.tsx (hunk 4 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/index.tsx:328]: What happens if there is no `to`?

File: packages/itinerary-body/src/TransitLegBody/index.tsx (hunk 3 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/index.tsx:325]: What happens if there is no `from`

**Pull request merged**